xGame/Java/xgame/src/main/java/com/xgame/service/UserService.java

```diff
@@ -1,35 +1,58 @@
 package com.xgame.service;
 
+import java.nio.charset.StandardCharsets;
 import java.util.List;
 import java.util.stream.Collectors;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
+import com.google.common.hash.Hashing;
 import com.xgame.common.viewmodels.ProfileViewModel;
 import com.xgame.common.viewmodels.UserViewModel;
 import com.xgame.data.IUserRepository;
+import com.xgame.data.entities.User;
 import com.xgame.service.interfaces.IUserService;
+import com.xgame.service.models.UserCredentials;
 
 @Service
 public class UserService implements IUserService {
 
 	@Autowired
 	private IUserRepository userRepo;
+	private final String salt = "boys-them-we-team";
 
 	@Override
-	public UserViewModel createNewUser(String name, String email, String password) {
-		// TODO Auto-generated method stub
-		return null;
+	public UserViewModel registerNewUser(UserCredentials credentials) {
+		var hashedPassword = Hashing.sha256().hashString(credentials.password + salt, StandardCharsets.UTF_8)
+				.toString();
+		var user = new User(credentials.nickname, credentials.email, hashedPassword);
+		userRepo.save(user);
+
+		return new UserViewModel(user);
 	}
-	
+
 	@Override
 	public List<UserViewModel> search(String param) {
 		var users = userRepo.findByNicknameLikeAndIsDeletedFalseOrEmailAndIsDeletedFalse("%" + param + "%", param);
-		
-		return users.stream()
-				.map(u -> new UserViewModel(u))
-				.collect(Collectors.toList());
+
+		return users.stream().map(u -> new UserViewModel(u)).collect(Collectors.toList());
+	}
+
+	@Override
+	public UserViewModel login(UserCredentials credentials) throws ResponseStatusException {
+		var hashedPassword = Hashing.sha256().hashString(credentials.password + salt, StandardCharsets.UTF_8)
+				.toString();
+		var user = userRepo.findByEmailAndPasswordHashAndIsDeletedFalse(credentials.email, hashedPassword);
```

Comment:
Since this should now return an `Optional<User>`, nullcheck with `Optional.isPresent()`